libraries/bdp-access-proxy/bdp-access-impl/src/main/java/com/bosch/osmi/bdp/access/impl/model/ProjectImpl.java

```diff
@@ -0,0 +1,126 @@
+import java.util.Map;
+
+/**
+ * @author muj1be
+ * @since 11/18/15.
+ */
+public class ProjectImpl implements Project {
+
+    private static final Logger LOGGER = LogManager.getLogger(ProjectImpl.class);
+
+    private final com.blackducksoftware.sdk.protex.project.Project project;
+    private final BdpApiAccessImpl access;
+
+    public ProjectImpl(com.blackducksoftware.sdk.protex.project.Project project, BdpApiAccessImpl access) {
+        this.project = project;
+        this.access = access;
+    }
+
+    @Override
+    public String getName() {
+        return project.getName();
+    }
+
+    @Override
+    public Collection<Component> getComponents() {
+        try {
+            List<BomComponent> bomComponents = access.getBomApi().getBomComponents(project.getProjectId());
+            Collection<Component> components = translate(bomComponents);
+            return components;
+        } catch (SdkFault sdkFault) {
+            sdkFault.printStackTrace();
+            LOGGER.error("Unable to retrieve components from Bdp server. Reason \n " + sdkFault.getMessage());
+            LOGGER.debug("Project name " + getName());
+            throw new IllegalStateException(sdkFault.getCause());
+        }
+    }
+
+    private Collection<Component> translate(List<BomComponent> bomComponents) {
+        Collection<Component> result = new ArrayList<Component>();
+        for (BomComponent source : bomComponents) {
+            result.add(new ComponentImpl(source, project.getProjectId(), access));
+        }
+        return result;
+    }
+
+    @Override
+    public String getCreatedBy() {
+        return project.getCreatedBy();
+    }
+
+    @Override
+    public String getFilesScanned() {
+        int Filecount = 0;
+        try {
+            AnalysisCodeTreeInfo analysisCodeTreeInfo = access.getDiscoveryApi()
+                    .getLastAnalysisCodeTreeInfo(project.getProjectId());
+            Filecount = analysisCodeTreeInfo.getAnalyzedFileCount();
+        } catch (SdkFault sdkFault) {
+            LOGGER.error("Unable to retrieve project data. Reason \n" + sdkFault.getMessage());
+            throw new IllegalStateException(sdkFault.getCause());
+        }
+        return String.valueOf(Filecount);
+    }
+
+    @Override
+    public String getPendingIdentification() {
+        long fileIdentifiedCount = 0;
+        String root = "/";
+        int treeDepth = 1;
+
+        try {
+            CodeTreeNodeRequest codeTreeParameters = new CodeTreeNodeRequest();
+            codeTreeParameters.getIncludedNodeTypes().addAll(CodeTreeUtilities.ALL_CODE_TREE_NODE_TYPES);
+            codeTreeParameters.setDepth(treeDepth);
+            codeTreeParameters.setIncludeParentNode(true);
+            codeTreeParameters.getCounts().add(NodeCountType.PENDING_ID_ALL);
+            List<CodeTreeNode> nodes = access.getCodeTreeApi().getCodeTreeNodes(project.getProjectId(), root, codeTreeParameters);
+
+            for (CodeTreeNode node : nodes) {
```

Comment:
This loop seems to implicitly assume that there can be only one node in `nodes`, or at least only one node that matters. Consider making this assumption explicit in code.